### DISCUSSION TITLE
A lean Scalar header to reduce compilation time

### DISCUSSION THREAD

**Discussion by zoharl3** [patr-schm/TinyAD#14] [General]
If anyone is interested, I wrote a tool to extract an interface from a C++ header to ease template instantiation for reducing compilation time. I used it on `Scalar.hh`:

https://github.com/zoharl3/minimize_cpp_header/tree/main/examples/TinyAD_Scalar

**Comment by patr-schm** [comment]
Hi Zohar,

awesome, this looks very useful! Thanks for posting it here.

I'd be curious whether you observed any changes in runtime performance due to less inlining.

**Reply by zoharl3** [reply]
Right now, I'm still at the stage where I care about compilation time. Maybe I'll update in the future. 

(And, of course, if there's significant overhead, then it's just a matter of changing which header is included in the "release" version.)

(I'm not sure if to close the discussion.)
